EventPlanner/app/src/main/java/m3/eventplanner/adapters/AgendaItemListAdapter.java

```diff
@@ -55,9 +77,40 @@ public void onBindViewHolder(@NonNull AgendaViewHolder holder, int position) {
 
         holder.nameTextView.setText(agendaItem.getName());
         holder.descriptionTextView.setText(agendaItem.getDescription());
-        holder.startTimeTextView.setText(agendaItem.getStartTime().substring(0, 5));
-        holder.endTimeTextView.setText(agendaItem.getEndTime().substring(0, 5));
+        holder.startTimeTextView.setText(agendaItem.getStartTime().toString());
+        holder.endTimeTextView.setText(agendaItem.getEndTime().toString());
         holder.locationTextView.setText(agendaItem.getLocation());
+        if(isOwner){
```

Comment:
Ensure that buttons are explicitly hidden for non-owners to avoid any unexpected behavior
if (isOwner) {
    holder.editButton.setVisibility(View.VISIBLE);
    holder.deleteButton.setVisibility(View.VISIBLE);
...
} else {
    holder.editButton.setVisibility(View.GONE);
    holder.deleteButton.setVisibility(View.GONE);
}